README.md

```diff
@@ -28,10 +26,132 @@ To generate organization-wide token:
 3. Select the application and choose generate token.
 4. Select `Administer organizations`, `Adminster repositories`, `Create Repositories` permissions.
 
-### Create a Component
+## General purpose image repository
+
+### Requesting image repository
+
+To request an image repository one should create `ImageRepository` custom resource:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ImageRepository
+metadata:
+  name: imagerepository-sample
+  namespace: test-ns
+```
+As a result, a public image repository `quay.io/my-org/test-ns/imagerepository-sample` will be created.
+When `status.state` is set to `ready`, the image repository is ready for use.
+Additional information about the image repository one may obtain from the `ImageRepository` object `status`:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ImageRepository
+metadata:
+  name: imagerepository-sample
+  namespace: test-ns
+spec:
+  image:
+    name: test-ns/imagerepository-sample
+    visibility: public
+status:
+  credentials:
+    generationTimestamp: "2023-08-23T14:56:41Z"
+    push-robot-account: test_ns_imagerepository_sample_101e4e2b63
+    push-secret: test-ns-imagerepository-sample-101e4e2b63
+  image:
+    url: quay.io/my-org/test-ns/imagerepository-sample
+    visibility: public
+  state: ready
+```
+where:
+  - `push-robot-account` is the name of quay robot account in the configured quay organization with write premissions to the repository.
+  - `push-secret` is a `Secret` of dockerconfigjson type that contains image repository access token with write permissions.
+
+### User defined image repository name
+
+One may request custom image repository name by setting `spec.image.name` field on the `ImageRepository` object creation.
+Note, it's not possible to change image repository name after creation.
+Any changes to the field will be reverted by the operator.
+
+---
+**NOTE**
+
+Image repository name is always prefixed with the `ImageRepository` namespace plus `/`.
+
+---
+
+### Image repository visibility
+
+It's possible to control image repository visibility by `spec.image.visibility` field.
+Allowed values are:
+ - `public` (default)
+ - `private`
+
+It's possible to set visibility on creation or any time later.
+
+---
+**NOTE**
+
+Your quay.io organization plan should allow creation of private repositories.
+In case your quay.io organization has free plan that does not allow setting repositories as private, then if `private` visibility was requested:
+ - on `ImageRepository` creation, then the creation will fail.
+ - after `ImageRepository` creation, then `status.message` will be set and `spec.image.visibility` reverted back to `public`.
+
+---
+
+### Credentials rotation
+
+It's possible to request robot account token rotation by adding:
+```yaml
+...
+spec:
+  ...
+  credentials:
+    regenerate-token: true
+  ...
+```
+After token rotation, the `spec.credentials` section will be deleted and `status.credentials.generationTimestamp` updated.
```

Comment:
Would be good to also mention whether or not and what else will be done accordingly

Reply:
I didn't get your point, could you elaborate more on this?

Reply:
I mean, except the `generationTimestamp` is updated, if anything else is also changed, it would be good to mention here as well.